st2common/st2common/runners/paramiko_ssh.py

```diff
@@ -579,13 +579,42 @@ def _connect(self, host, socket=None):
         if socket:
             conninfo['sock'] = socket
 
+        if cfg.CONF.ssh_runner.use_ssh_config:
+            conninfo_host_ssh_config = self._get_conninfo_from_ssh_config_for_host(host)
```

Comment:
Overall the PR looks good to me, but to make debugging easier, please add a log statement which says something along the lines of "Using SSH config file from <path>".

And perhaps also log which options (with values) are overridden from the config file.